plotting-visualization/rtdose_visual.py

```diff
@@ -0,0 +1,199 @@
+# -*- coding: utf-8 -*-
+"""
+Created on Wed Mar 31 10:45:06 2021
+
+@author: Francisco Piqueras Guardiola
+@author: Fernando Hueso González
+"""
+import pydicom
+import matplotlib.pyplot as plt
+import os
+import numpy as np
+#!/usr/bin/python
+
+# Define the path folder where the file is and the global variables
+path = ''
+dirs = os.listdir( path )
+global gaxis
+gaxis= 0 %2
+global cMin, cMax,cMap
+cMap = 'turbo'
+cMin = None
+cMax = None
+
+#Here we define the functions that we are gona use in our program
+def remove_keymap_conflicts(new_keys_set):
+    """
+    Removes possible conflicts created by the use of a key that already has an use in the plot.
+
+    Parameters
+    ----------
+    new_keys_set : set
+
+    Returns
+    -------
+    None.
+
+    """
+    for prop in plt.rcParams:
+        if prop.startswith('keymap.'):
+            keys = plt.rcParams[prop]
+            remove_list = set(keys) & new_keys_set
+            for key in remove_list:
+                keys.remove(key)
+   
+def multi_slice_viewer(volume):
+    """
+    Based on : https://www.datacamp.com/community/tutorials/matplotlib-3d-volumetric-data
```

Comment:
What is the licensing on that code?  If not MIT-compatible, then should not be using any code directly (not sure what "based on" implies here).

Reply:
It's not a source code directly, it's just some snippets within a Python visualization tutorial. So not sure if a license applies at all. They do not say anything explicitly in that post.

With "Based" we mean that some of these functions are essentially the same, with some minor modifications to allow changing the axis of view, not just the slice number.

Reply:
Can you check a little further?  I think a function that is essentially the same would require the license to be compatible with pydicom.  The repo at https://github.com/datacamp/datacamp-community-tutorials, with a few tutorials, is MIT-license, but I don't know if that implies that all are.  I would expect datacamp should have a statement somewhere about tutorial code license if not otherwise specified, but perhaps the author can be contacted if not.

Reply:
Thanks for the directions. I will contact the author.

Reply:
He answered, he says the original code is available here https://github.com/jni/mpl-volume-viewer/blob/master/LICENSE under a BSD-3, so he said we can reuse it without problems.

Reply:
👋 Hello! I responded to @ferdymercury privately but since I got the link here:

The code is fully available here:

https://github.com/jni/mpl-volume-viewer

and is BSD 3-clause. Enjoy! ;)

Reply:
@kikooku or @ferdymercury, I suggest then that the "based on" link in this PR's code be updated to point to the BSD location.  With that and removal of `in_py2`, we can get this merged in.

Reply:
We'll do that, thanks. We are cleaning up a couple of things in the code also.

Reply:
Done